scripts/run_all_checks.py

```diff
@@ -0,0 +1,63 @@
+import json
+import re
+import subprocess
+import sys
+
+# Popular Flake8 extensions not used in Typeshed that cause errors
+_extend_ignore = "CCE,N8,Q"
+
+_STRICTER_CONFIG_FILE = "pyrightconfig.stricter.json"
+
+
+def _parse_jsonc(json_text: str) -> str:
+    parsed_json_text = re.sub(r"^\s*?//.*?\n", "", json_text, flags=re.RegexFlag.MULTILINE)
+    parsed_json_text = re.sub(r",([\n\s]*?[\}\]])", r"\1", parsed_json_text)
+    return parsed_json_text
+
+
+def _get_strict_params(stub_path: str) -> list[str]:
+    with open(_STRICTER_CONFIG_FILE) as file:
+        data = json.loads(_parse_jsonc(file.read()))
+    if stub_path in data["exclude"]:
+        return []
+    return ["-p", _STRICTER_CONFIG_FILE]
+
+
+if __name__ == "__main__":
+    try:
+        path = sys.argv[1]
+    except IndexError:
+        print("Missing path argument in format <folder>/<stub>")
+        sys.exit(1)
+    path_tokens = path.replace("\\", "/").split("/")
+    assert len(path_tokens) == 2, "Path argument should be in format <folder>/<stub>"
+    folder, stub = path_tokens
+    assert folder in {"stdlib", "stubs"}, "Only the 'stdlib' and 'stubs' folders are supported"
+
+    subprocess.run(["python3", "-m", "black", path])
+    subprocess.run(["python3", "-m", "isort", path])
+    subprocess.run(["python3", "-m", "flake8", path, "--extend-ignore", _extend_ignore])
+
+    subprocess.run(["python3", "tests/check_consistent.py"])
+    subprocess.run(["python3", "tests/check_new_syntax.py"])
+
+    subprocess.run(["python3", "tests/pyright_test.py", path] + _get_strict_params(path))
```

Comment:
I'd prefer it if this didn't fail if pyright's not installed. E.g. I don't have pyright installed locally in my typeshed environment -- I've tried once or twice, but it didn't work immediately, I didn't bother to investigate why, and relying on pyright in CI is good enough for me ;)

Reply:
Are you asking this *specifically* because pyright may be slightly different than others? (uses node and is installed in the global `node_modules` instead of python in a virtual env).

If pyright fails to run (not installed, npx not found, etc.), you already get a proper message from `pyright_test.py` and it doesn't stop the execution.

See how I can't run mypy locally for example (because of the mentioned issue), but this entire script still completes.

I don't really have a way of differentiating between "failed pyright test" and "pyright failed to run" that I can think of. `shutil.which("pyright")` doesn't make sense if ran through npx, and I don't think a `shutil.which("npx")` check would change anything for you.

Reply:
> Are you asking this _specifically_ because pyright may be slightly different than others?

Yeah, pretty much — if you run `pip install -r requirements-tests.txt`, it's the only one that won't _necessarily_ be installed locally.

> I don't really have a way of differentiating between "failed pyright test" and "pyright failed to run" that I can think of. `shutil.which("pyright")` doesn't make sense if ran through npx, and I don't think a `shutil.which("npx")` check would change anything for you.

Fair enough :) I haven't actually tried running this script yet — I'll give it a go soon and see if I have any specific suggestions

Reply:
Can you give me the output (stderr) you get when running `pyright_test.py`? Since I now show which tests are skipped because they can't be ran locally, (and I learned how to pipe the output), I can check for npx and whatever error you get.

Reply:
The output printed to stderr on my machine is:

```
error finding npx; is Node.js installed?\n
```